### PR TITLE
mgr/cephadm: Enable provisioning alertmanager via orchestrator

### PR DESCRIPTION
**Pull request by krig**
Configures the alertmanager to send alerts to the dashboard.

Tracker ticket: https://tracker.ceph.com/issues/44308

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by krig** [comment]
@pcuzner Since you implemented the cephadm part, maybe you could take a look at this? Thank you :)

**Comment by liewegas** [comment]
- can you squash the fixes in the last commit into the relevant commits?
- updating prom config sounds good.

note that a pending TODO is to make this happen automatically (e.g., after a new alertmanager is deployed the prom config is magically updated).  that can come later

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2336,6 +2339,43 @@ def create_self_signed_cert() -> Tuple[str, str]:
         })
         return config_file
 
+    def _get_dashboard_url(self):
+        # type: () -> str
```
Comment (jschmid1) [inline src/pybind/mgr/cephadm/module.py]: Wouldn't that return either `str` or `None`?

Comment (krig) [reply src/pybind/mgr/cephadm/module.py]: ah yes, true

**Comment by krig** [comment]
@p-se @jschmid1 @liewegas All the comments should be addressed now, thanks!

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -2241,6 +2265,7 @@ def _generate_prometheus_config(self):
 """.format(
     mgr_scrape_list=str(mgr_scrape_list),
     node_configs=str(node_configs),
+    alertmgr_configs=str(alertmgr_configs)
```
Comment (p-se) [inline src/pybind/mgr/cephadm/module.py]: `alertmgr_configs` is an empty string at this point

**Comment by p-se** [comment]
The make check issue is unrelated to this PR. There was an issue in master which has been fixed by https://github.com/ceph/ceph/pull/33573.

File: src/pybind/mgr/cephadm/module.py (hunk 2 above)
Comment (krig) [reply src/pybind/mgr/cephadm/module.py]: yep, it’s a typo, I’ll fix it

**Pull request merged**